docs/_snippets/framework/tutorials/external-widget.js

```diff
@@ -168,24 +163,21 @@ class ExternalWidgetEditing extends Plugin {
 		editor.conversion.for( 'editingDowncast' ).elementToElement( {
 			model: 'externalElement',
 			view: ( modelElement, { writer: viewWriter } ) => {
-				const externalUrl = modelElement.getAttribute( 'data-resource-url' );
-				const fetchAndUpdateData = this._updateWidgetDataFn;
-
-				const viewContentWrapper = viewWriter.createRawElement( 'span', {
-					class: 'js-external-data-embed'
-				}, async function( domElement ) {
-					domElement.textContent = 'Fetching data...';
+				const externalValueToShow = this.externalDataValue;
 
-					const data = await fetchAndUpdateData( true, externalUrl );
+				const externalDataPreviewElement = viewWriter.createRawElement( 'span', null, function( domElement ) {
+					domElement.textContent = externalValueToShow;
 
-					domElement.textContent = data;
+					if ( externalValueToShow !== 'Fetching data...' ) {
```

Comment:
After initializing `externalDataValue` property string (as I mentioned in another comment):

```suggestion
					if ( externalValueToShow ) {
```